### DISCUSSION TITLE
How to close or delete Preview Deployments ?

### DISCUSSION THREAD

**Discussion by gyhyfj** [vercel/vercel#8165] [Help] [answered]
There is a long list on status `ready`, but they're really no nessary

**Comment by amyegan** [comment]
You can delete individual deployments from the Deployments tab of your Project page. Step-by-step guide here 👉 https://vercel.com/support/articles/how-do-i-delete-an-individual-deployment

**Comment by rashidul0405** [comment]
@amyegan  How do I automate this? For example, As an admin, when the pull request/branch merged into the main branch, it should automatically delete the preview deployment associated with that pull request.

**Reply by amyegan** [reply]
Hi @rashidul0405! You can delete preview deployments via Vercel CLI using [`vercel remove`](https://vercel.com/docs/cli/remove#) or via Vercel API at [`/v13/deployments/{id}`](https://vercel.com/docs/rest-api#endpoints/deployments/delete-a-deployment). The team is already planning additional options for managing/removing stale deployments, but for now you could use the CLI with GitHub Actions for automatic preview deployment deletion when a PR is merged.

**Reply by thereis** [reply]
I've created a blog post explaining how I achieved this in automatic way using github actions. This can be very helpful for newcomers.

https://dev.to/thereis/how-to-remove-vercel-deployments-from-github-actions-4nfh

**Comment by leerob** [comment, marked as answer]
You can now configure the retention period for deployments through the dashboard and CLI.

For example, canceled and errored deployments might be set to 30 days retention, while production deployments might be set to 1 year. Recently deleted deployments are shown in your project settings and can be instantly restored within 30 days of deletion.

Learn more in our [documentation](https://vercel.com/docs/security/deployment-retention).
